### PR TITLE
[mod_mariadb] Fix reconnect during db failover/restart

### LINKED ISSUE #522: Bug: mod_mariadb connection cannot reconnect during db failover/restart
If you have a DB cluster (maybe also with a single server) - the mod_mariadb reconnect code does not work. It is necessary to set the MYSQL_OPT_RECONNECT setting in mysql_options.

```
2020-03-16 19:49:54.530062 [WARNING] mod_mariadb.c:197 mysql_ping returned bad connection [Error: MySQL server has gone away]; reconnecting...
2020-03-16 19:49:54.530062 [CRIT] mod_mariadb.c:200 mariadb_reconnect returned bad connection -- reconnection failed! [Error: MySQL server has gone away]
```

(along with a flooded log message that includes database credentials... that's a separate issue I'd really like to see improved)

### PR DESCRIPTION
**Pull request by nneul**

### CODE REVIEW

File: src/mod/databases/mod_mariadb/mod_mariadb.c [hunk 1]
```diff
@@ -478,6 +480,9 @@ switch_status_t mariadb_handle_connect(mariadb_handle_t *handle)
 	// https://mariadb.com/kb/en/library/using-the-non-blocking-library/
 	mysql_options(&handle->con, MYSQL_OPT_NONBLOCK, 0);
 
+	// Enable automatic reconnect with the mariadb_reconnect function, without this that function does not work
```
Comment (andywolk) [inline src/mod/databases/mod_mariadb/mod_mariadb.c]: Minor, but please use `/* */` comment style

Comment (nneul) [reply src/mod/databases/mod_mariadb/mod_mariadb.c]: ok - was just matching the existing syntax two lines above it.... but happy to change.

Comment (nneul) [reply src/mod/databases/mod_mariadb/mod_mariadb.c]: updated.

**Comment by andywolk** [comment]
Thank you for the contribution!

**Pull request merged**